codalab/server/bundle_manager.py

```diff
@@ -689,14 +689,13 @@ def _check_resource_failure(
     def _get_matched_workers(request_queue, workers):
         """
         Get all of the workers that match with the name of the requested worker
-        :param request_queue: a tag that can be used to match workers
+        :param request_queue: a tag with the format "tag=worker_X" or "worker_X" that can be used to match workers
         :param workers: a list of workers
         :return: a list of matched workers
         """
-        tag_match = re.match('tag=(.+)', request_queue)
-        if tag_match != None:
-            return [worker for worker in workers if worker['tag'] == tag_match.group(1)]
-        return []
+        tag_match = re.match('(?:tag=)?(.+)', request_queue)
+        worker_tag = tag_match.group(1) if tag_match != None else None
+        return [worker for worker in workers if worker['tag'] == worker_tag]
```

Comment:
Does this list comprehension work when `worker_tag == None` ?

Reply:
When `worker_tag == None`, if there is no `worker['tag']` equals `None`, it will return an empty list. One thing I just realized is that what if there are workers with `tag==None`, then they will be selected in this logic, which might not be correct. I printed out the worker object as follows:
```
workers = [{'user_id': '0', 
'worker_id': 'Stanfords-MacBook-Pro.local', 
'tag': None, 
'cpus': 4, 
'gpus': 0, 
...,
'shared_file_system': False}]
``` 
It appears that a worker without a non-empty `tag` contains the value `None`, which might cause problems in matching workers. I changed the `worker_tag` to be an empty string if no matching workers could be found. This should be safe then.